src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -257,6 +265,36 @@ void InferRequestBase::changeDefaultPtr() {
         auto output = outputNodesMap.find(it.first);
         if (output != outputNodesMap.end()) {
             auto parentEdge = output->second->getParentEdgeAt(0);
+            if (Graph::Status::ReadyDynamic == graph->getStatus()) {
+                bool canBeInPlace = true;
+                // TODO: filter
+
+                auto inferrequest = std::dynamic_pointer_cast<InferRequest>(this->shared_from_this());
+                OPENVINO_ASSERT(inferrequest, "should be a InferRequest instance");
+
+                // share intel_cpu::Tensor to Graph by injecting to corresponding OutputMemoryMngr instance.
+                OutputMemoryMngrPtr outputMemMngr;
+                const auto &outMemMngrMap = graph->outputNodesMemMngrMap;
+                auto itr = outMemMngrMap.find(it.first);
+                if (itr != outMemMngrMap.end()) {
+                    outputMemMngr = std::dynamic_pointer_cast<OutputMemoryMngr>(itr->second);
+                    OPENVINO_ASSERT(outputMemMngr, "output memmanager should not be empty.");
+                } else {
+                    OPENVINO_THROW("Cannot find output memmanager for output " + it.first + " !");
+                }
+
+                if (canBeInPlace) {
+                    auto tt = std::get<0>(outputsTensor2BlobMap[it.first]);
+                    outputMemMngr->setTensor(tt);
+                    DEBUG_LOG("setTensor ", tt, " graph ", graph, " inferrequest ", this);
+                } else {
+                    outputMemMngr->setTensor(nullptr);
```

Comment:
There is a memory allocation in graph init stage for output edges. In situations that zero-copy is not feasible, we could reuse that memory manager, instead of create a new instance. How do you think?